Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -230,14 +257,297 @@ ContourExtractor2DImageFilter<TInputImage>::GenerateData()
+    IndexType       min = right_bot;
+    {
+      RegionConstIterator inputIt{ this->GetInput(), inputRegion };
+      for (inputIt.GoToBegin(); !inputIt.IsAtEnd(); ++inputIt)
+      {
+        if (inputIt.Get() == label)
+        {
+          max[0] = std::max(max[0], inputIt.GetIndex()[0]);
+          max[1] = std::max(max[1], inputIt.GetIndex()[1]);
+          min[0] = std::min(min[0], inputIt.GetIndex()[0]);
+          min[1] = std::min(min[1], inputIt.GetIndex()[1]);
+        }
+      }
+      // Set boundary values in largerRegion to be distinct from label if they will be looked at.
+      if (min[0] == left_top[0])
+      {
+        const IndexType       stripeIndex{ min[0] - 1, min[1] - 1 };
+        const SizeType        stripeSize{ 1, static_cast<SizeValueType>(max[1] - min[1] + 3) };
+        const InputRegionType stripeRegion{ stripeIndex, stripeSize };
+        RegionIterator        stripeIt{ largerImage, stripeRegion };
+        const InputPixelType  differentLabel = label + 1;
+        for (stripeIt.GoToBegin(); !stripeIt.IsAtEnd(); ++stripeIt)
+        {
+          stripeIt.Set(differentLabel);
+        }
+      }
+      if (min[1] == left_top[1])
+      {
+        const IndexType       stripeIndex{ min[0] - 1, min[1] - 1 };
+        const SizeType        stripeSize{ static_cast<SizeValueType>(max[0] - min[0] + 3), 1 };
+        const InputRegionType stripeRegion{ stripeIndex, stripeSize };
+        RegionIterator        stripeIt{ largerImage, stripeRegion };
+        const InputPixelType  differentLabel = label + 1;
+        for (stripeIt.GoToBegin(); !stripeIt.IsAtEnd(); ++stripeIt)
+        {
+          stripeIt.Set(differentLabel);
+        }
+      }
+      if (max[0] == right_bot[0])
+      {
+        const IndexType       stripeIndex{ max[0] + 1, min[1] - 1 };
+        const SizeType        stripeSize{ 1, static_cast<SizeValueType>(max[1] - min[1] + 3) };
+        const InputRegionType stripeRegion{ stripeIndex, stripeSize };
+        RegionIterator        stripeIt{ largerImage, stripeRegion };
+        const InputPixelType  differentLabel = label + 1;
+        for (stripeIt.GoToBegin(); !stripeIt.IsAtEnd(); ++stripeIt)
+        {
+          stripeIt.Set(differentLabel);
+        }
+      }
+      if (max[1] == right_bot[1])
+      {
+        const IndexType       stripeIndex{ min[0] - 1, max[1] + 1 };
+        const SizeType        stripeSize{ static_cast<SizeValueType>(max[0] - min[0] + 3), 1 };
+        const InputRegionType stripeRegion{ stripeIndex, stripeSize };
+        RegionIterator        stripeIt{ largerImage, stripeRegion };
+        const InputPixelType  differentLabel = label + 1;
+        for (stripeIt.GoToBegin(); !stripeIt.IsAtEnd(); ++stripeIt)
+        {
+          stripeIt.Set(differentLabel);
+        }
+      }
+    }
+    const IndexType       shrunkIndex{ min[0] - 1, min[1] - 1 };
+    const SizeType        shrunkSize{ static_cast<SizeValueType>(max[0] - min[0]) + 2,
```

Comment:
Couldn't we reuse `GenerateDataForIntensities()` here?

Reply:
Possibly.  If we had a common subroutine that accepted the shrunk region, etc. and accepted a function (or function object) that could do it.  The function object would handle the difference between https://github.com/InsightSoftwareConsortium/ITK/blob/c8daf734c856ba52f0f19b605bf8249ea9ccc3e5/Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx#L155-L158 and https://github.com/InsightSoftwareConsortium/ITK/blob/c8daf734c856ba52f0f19b605bf8249ea9ccc3e5/Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx#L459-L462.

Reply:
This is now Issue #2349.